docs/source/llms/langchain/index.rst

```diff
@@ -156,110 +164,41 @@ exploring these more advanced use cases.
---------------------------------------------
-
-To learn more about the details of the MLflow LangChain flavor, read the detailed guide below.
 
-.. raw:: html
+Logging models from Code
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-    <a href="guide/index.html" class="download-btn">View the Comprehensive Guide</a>
+Since MLflow 2.12.2, MLflow introduced the ability to log LangChain models directly from a code definition.
 
-.. toctree::
-    :maxdepth: 1
-    :hidden:
+The feature provides several benefits to manage LangChain models:
 
-    guide/index.rst
-
-FAQ
----
-
-I can't load my chain!
-^^^^^^^^^^^^^^^^^^^^^^
-
-- **Allowing for Dangerous Deserialization**: Pickle opt-in logic in LangChain will prevent components from being loaded via MLflow. You might see an error like this:
-
-    .. code-block:: text
+1. **Avoid Serialization Complication**: File handles, sockets, external connections, dynamic references, lambda functions and system resources are unavailable for pickling. Some LangChain components does not support native serialization, e.g. ``RunnableLambda``.
 
-        ValueError: This code relies on the pickle module. You will need to set allow_dangerous_deserialization=True if you want to opt-in to 
-        allow deserialization of data using pickle. Data can be compromised by a malicious actor if not handled properly to include a malicious 
-        payload that when deserialized with pickle can execute arbitrary code on your machine. 
+2. **No Pickeling**: Loading a pickle or cloudpickle file in a Python version that was different than the one used to serialize the object does not guarantee compatibility.
 
-    A change within LangChain that `forces users to opt-in to pickle deserialization <https://github.com/langchain-ai/langchain/pull/18696>`_ can create 
-    some issues with loading chains, vector stores, retrievers, and agents that have been logged using MLflow. Because the option is not exposed per component
-    to set this argument on the loader function, you will need to ensure that you are setting this option directly within the defined loader function when 
-    logging the model. LangChain components that do not set this value will be saved without issue, but a ``ValueError`` will be raised when loading if unset. 
+3. **Readability**: The serialized objects are often hardly readable by humans. Model-from-code allows you to review your model definition via code.
 
-    To fix this, simply re-log your model, specifying the option ``allow_dangerous_deserialization=True`` in your defined loader function. See the tutorial 
-    `for LangChain retrievers <notebooks/langchain-retriever.html#Establishing-RetrievalQA-Chain-and-Logging-with-MLflow>`_ for an example of specifying this
-    option when logging a ``FAISS`` vector store instance within a ``loader_fn`` declaration.
 
+Refer to the `Models From Code feature documentation <../../models.html#models-from-code>`_ for more information about this feature.
```

Comment:
We'd better add a note here and in models-from-code page  saying that: 
Please check your langchain model definition code doesn't include credentials such as Openai token, MLFlow loggging implementation doesn't check it and it might cause your private token being uploaded to MLflow model repository unexpectedly.

Reply:
100%, added a warning block to the both page!